lib/core/presentation/pages/event/event_detail_page/host_event_detail_page/sub_pages/scan_qr_checkin_rewards/views/scan_qr_checkin_rewards_view.dart

```diff
@@ -48,26 +48,49 @@ class _ScanQRCheckinRewardsViewState extends State<ScanQRCheckinRewardsView> {
 
   Future<void> onBarcodeDetect(BarcodeCapture barcodeCapture) async {
     await controller.stop();
-    final userId = barcodeCapture.barcodes.isNotEmpty
+    final shortId = barcodeCapture.barcodes.isNotEmpty
         ? barcodeCapture.barcodes.last.displayValue?.toString() ?? ''
         : '';
     if (widget.selectedScannerTabIndex == SelectedScannerTab.checkIn.index) {
-      context.read<UpdateEventCheckinBloc>().add(
-            UpdateEventCheckinEvent.checkinUser(
-              active: true,
-              eventId: widget.event.id ?? '',
-              userId: userId,
+      final response = await showFutureLoadingDialog(
+        context: context,
+        future: () => getIt<AppGQL>().client.mutate$UpdateEventCheckin(
+              Options$Mutation$UpdateEventCheckin(
+                variables: Variables$Mutation$UpdateEventCheckin(
+                  input: Input$UpdateEventCheckinInput(
+                    event: widget.event.id ?? '',
+                    active: true,
+                    shortid: shortId,
+                  ),
+                ),
+              ),
             ),
-          );
+      );
+      if (response.result?.parsedData?.updateEventCheckin != null) {
+        SnackBarUtils.showSuccess(
+          message: t.event.scanQR.checkedinSuccessfully,
+        );
+        await AutoRouter.of(context).pop();
+      }
     } else if (widget.selectedScannerTabIndex ==
         SelectedScannerTab.rewards.index) {
-      await AutoRouter.of(context).navigate(
-        ClaimRewardsRoute(userId: userId),
+      final response = await showFutureLoadingDialog(
+        context: context,
+        future: () =>
+            getIt<EventTicketRepository>().getTicket(shortId: shortId),
+      );
+      response.result?.fold(
+        (failure) => null,
+        (ticket) async {
+          final assignedTo = ticket.assignedTo;
+          if (assignedTo != null) {
```

Comment:
should we inform user incase no assigned to ? 
In Ticket domain, there are two field `assigned_to` and `assigned_email`, you might need to check with BE on this for non-loggedin user cases where assigned_to is null, but assigned_email is available, double check if we can claim the reward with email for non-login user?